core/app/models/concerns/spree/user_address_book.rb

```diff
@@ -116,17 +136,30 @@ def save_in_address_book(address_attributes, default = false)
         # user_addresses need to be reset to get the new ordering based on any changes
         # {default_,}user_address needs to be reset as its result is likely to have changed.
         user_addresses.reset
-        association(:default_user_address).reset
-        association(:default_address).reset
+        association(:default_user_ship_address).reset
+        association(:ship_address).reset
+        association(:default_user_bill_address).reset
+        association(:bill_address).reset
       end
 
       user_address.address
     end
 
     def mark_default_address(address)
+      Spree::Deprecation.warn "Hey, this method is deprecated and it sets the ship_address only! " \
```

Comment:
I know I initially suggested how to formulate this sentence, but `Hey, ` wasn't meant to end up in real code. :P 
Can you please restyle this message in line with the tone of the other deprecations (just a little bit more formal).

Reply:
Done